packages/suite/src/views/onboarding/steps/ResetDevice.tsx

```diff
@@ -1,111 +1,154 @@
+        },
+        [dispatch, goToNextStep],
+    );
 
-    const handleSingleSeedReset = async () => {
-        if (isShamirBackupAvailable) {
-            await onResetDevice({ backup_type: 0 });
-        } else {
-            await onResetDevice();
-        }
+    const handleSubmit = useCallback(
+        async (type: BackupType) => {
+            switch (type) {
+                case 'shamir-default':
+                    await onResetDevice({ backup_type: 1 }); // Todo: add number of shards = 1/1
+                    break;
+                case 'shamir-advance':
+                    await onResetDevice({ backup_type: 1 }); // Todo: add number of shards = n/m (select on device?)
+                    break;
+                case '12-words':
+                    await onResetDevice({ backup_type: 0, strength: 128 });
+                    break;
+                case '24-words':
+                    await onResetDevice({ backup_type: 0, strength: 256 });
+                    break;
+            }
 
-        updateAnalytics({ recoveryType: undefined, seedType: 'standard' });
-    };
+            updateAnalytics({ recoveryType: undefined, seedType: type });
+        },
+        [updateAnalytics, onResetDevice],
+    );
 
-    const handleShamirReset = async () => {
-        await onResetDevice({ backup_type: 1 });
+    useEffect(() => {
+        if (device_model !== undefined && !canChooseBackupType(device_model)) {
+            handleSubmit(defaultBackupTypeMap[device_model]);
+        }
+    }, [device_model, handleSubmit]);
 
-        updateAnalytics({ recoveryType: undefined, seedType: 'shamir' });
-    };
+    // this step expects device
+    if (!device || !device.features) {
+        return null;
+    }
 
     return (
         <OnboardingStepBox
             image="KEY"
             heading={<Translation id="TR_ONBOARDING_GENERATE_SEED" />}
-            description={<Translation id="TR_ONBOARDING_GENERATE_SEED_DESCRIPTION" />}
+            description={
+                device_model !== undefined && canChooseBackupType(device_model) ? (
+                    <Translation
+                        id="TR_ONBOARDING_SELECT_SEED_TYPE"
+                        values={{
+                            primary: chunks => <Text variant="primary">{chunks}</Text>,
+                        }}
+                    />
+                ) : (
+                    <Translation id="TR_ONBOARDING_CANNOT_SELECT_SEED_TYPE" />
+                )
+            }
             device={isWaitingForConfirmation ? device : undefined}
             isActionAbortable={isActionAbortable}
             outerActions={
-                !isWaitingForConfirmation ? (
+                !isWaitingForConfirmation && !isDeviceLocked ? (
                     // There is no point to show back button if user can't click it because confirmOnDevice bubble is active
                     <OnboardingButtonBack onClick={() => goToPreviousStep()} />
                 ) : undefined
```

Comment:
nitpick: there is no need for ternary operator with undefined as a second option, you can use `&&`